### PR TITLE
Content metadata redux

### PR DESCRIPTION
**Pull request by i-like-robots**
Re-factored content V1 metadata transformations. 

The current setup is some of the oldest code we have and due its 'organic' growth and unexpected longevity it's currently [rather scary](https://github.com/Financial-Times/next-es-interface/blob/c3138578947fb2e0ea2bd93e9f47a8dc3f341ada/server/lib/metadata-transformer.js).

This hopefully makes the code much more approachable and comes with easy-to-read specs \0/

Due to the large size of this PR I will do the clean up in a separate branch.

I've verified the metadata output against the current implementation with the following script (changes to primary section/theme are _correct_):

```js
require('isomorphic-fetch');

const { series } = require('promise-patterns');
const diff = require('deep-diff');

const ids = [
	// life & arts
	'9d568bb6-9010-11e6-8df8-d3778b55a923',
	// theme
	'a1527120-9511-11e6-a80e-bcd69f323a8b',
	// people
	'7edf0078-9218-11e6-8df8-d3778b55a923',
	// opinion
	'96939c42-944d-11e6-a80e-bcd69f323a8b',
	// fast ft
	'429a4108-63d8-36b7-8435-8b242f057e20',

	// a few more randoms...
	'6af9756e-3b30-11e1-b7ba-00144feabdc0',
	'e2beac6c-f703-11df-8feb-00144feab49a',
	'9b29866c-019f-3113-841c-a238115125e9',
	'871a6754-0e3d-3204-b1ec-aa61b39da820',
	'97a9a154-f866-11e0-a419-00144feab49a',
	'aa9276d0-8990-11dd-8371-0000779fd18c',
	'b2dd04e4-498e-11e4-80fb-00144feab7de',
	'b58bd01a-9e3c-11de-b0aa-00144feabdc0',
	'043f277c-1963-11e1-b490-00144feabdc0',
	'caa34fa4-1a23-11e1-9888-00144feabdc0',
	'b66f8870-ff08-3b5a-9c5d-ed45ce623e8a',
	'de083a58-1539-11e1-b9b8-00144feabdc0',
	'7f1a158c-b71f-11e2-a249-00144feabdc0',
	'd7ae68a4-17cc-3c4a-b125-979cd6d97a84',
	'20a7d046-a8ee-11e2-a096-00144feabdc0',
	'd2d6ea14-b342-11e2-95b3-00144feabdc0',
	'cbaa45c4-486c-11e2-a1c0-00144feab49a',
	'8e02d198-2ef9-11e5-8873-775ba7c2ea3d',
	'1c843d80-accf-11e2-9454-00144feabdc0',
	'd6ec7330-f087-11e2-929c-00144feabdc0',
	'94e3cd0f-3c62-3d79-b014-7c56128cb0c5',
	'530d8bf8-e64a-11e3-9a20-00144feabdc0',
	'ff0f3245-b503-337f-8297-33916b4b243c',
	'7af73c2e-930b-11e2-b3be-00144feabdc0',
	'771f4e8e-f766-11e1-8e9e-00144feabdc0',
	'582f8ef6-19ff-11e2-a379-00144feabdc0',
	'afb69c90-7dfa-11e3-b409-00144feabdc0',
	'ab15a4ba-b30c-11e2-95b3-00144feabdc0',
	'1e31583c-0b87-11e2-8e06-00144feabdc0',
	'70fae800-3e25-11e2-91cb-00144feabdc0',
	'b74123ec-7f26-11e1-a26e-00144feab49a',
	'74fa2cd7-b2f0-32f5-90b6-bd7890a0dfa5',
	'74ba086c-ecca-11e4-b82f-00144feab7de',
	'ad0b2ee6-b318-11e2-95b3-00144feabdc0',
	'4c130544-e0fa-11e3-875f-00144feabdc0',
	'2194b5c8-adad-11e2-a2c7-00144feabdc0',
	'8303666a-ac3b-11e2-a063-00144feabdc0',
	'41a28b6c-affe-11e2-8d07-00144feabdc0',
	'c8cd8c90-be45-11e2-bb35-00144feab7de',
	'59fdd970-9a35-11e2-9732-00144feabdc0',
];

const get = (href) => (
	fetch(href).then((res) => {
		if (res.ok) {
			return res.json();
		} else {
			throw new Error(`${href} responded with a ${res.status}`);
		}
	})
);

const sort = (a, b) => a.idV1.localeCompare(b.idV1);

const requests = ids.map((id) => {
	return function () {
		return Promise.all([
			get(`http://localhost:3002/api/preview/${id}?apiKey=KEY`),
			get(`http://ft-next-es-interface-eu.herokuapp.com/api/preview/${id}?apiKey=KEY`)
		])
			.then((results) => {
				const lhs = results[0].metadata.sort(sort);
				const rhs = results[1].metadata.sort(sort);

				// we will always add an empty array
				const changes = diff(lhs, rhs, (path, key) => key === 'attributes');

				if (changes) {
					console.error(`${id} is different`);
					console.dir(Array.isArray(changes) ? changes[0] : changes);
				} else {
					console.log(`${id} is equal`);
				}
			})
			.catch((err) => console.error(err));
	}
});

series(requests);
```

### CODE REVIEW

File: server/models/article/metadata/extract-tags.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const predicates = {
+	object: ['primaryTheme', 'primarySection'],
+	array: [
+		'tags', 'authors', 'brand', 'genre', 'icb', 'iptc',
+		'mediaType', 'organisations', 'people', 'regions',
+		'sections', 'specialReports', 'subjects', 'topics'
+	]
+};
```
Comment (i-like-robots) [inline server/models/article/metadata/extract-tags.js]: If the `tags` array is not required then I can ignore the unwanted taxonomies here and skip the extra step to remove them.

File: test/server/models/article/metadata/display-tags-spec.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	name: 'News',
+	taxonomy: 'genre'
+};
+
+describe('Metadata - Display tags', () => {
+	let fixture;
+	let primaryTag;
+	let teaserTag;
+
+	// it('returns an array', () => {
```
Comment (andygnewman) [inline test/server/models/article/metadata/display-tags-spec.js]: Some commented out code here, can it be removed?

**Comment by i-like-robots** [comment]
Thanks for checking - my comment to remind me to check this morning but due to some ES troubles I haven't yet.

**Comment by i-like-robots** [comment]
Checks are good, I've removed the taxonomies filter in favour of not plucking them out in the first place.

**Pull request merged**